app/engine/engine.ts

```diff
@@ -112,6 +113,34 @@ export function gameEngine(state: State, action: Action): State {
           draft.boardControl = action.payload.userId;
         }
       });
+    case ActionType.Kick:
+      if (!isPlayerAction(action)) {
+        throw new Error("PlayerKick action must have an associated player");
+      }
+      return produce(state, (draft) => {
+        // Don't kick players after the game has started.
+        if (
+          (draft.type !== GameState.ShowBoard &&
+            draft.type !== GameState.PreviewRound) ||
+          draft.numAnswered > 0 ||
+          draft.round > 0
+        ) {
+          return;
+        }
+        // Don't allow the only player to be kicked.
+        if (draft.players.size === 1) {
```

Comment:
nice catch, can you prevent this on the client side as well (once players can kick themselves)?